ska_dbi/sqsh.py

```diff
@@ -0,0 +1,168 @@
+
+    :param server: Server name (default = sqlsao)
+    :param user: User name (default = aca_ops)
+    :param passwd: Password (optional).  Read from aspect authorization dir if required and not supplied.
+    :param database: Database name (default = axafapstat)
+    :param sqshrc: sqshrc file (optional).  Read from aspect authorization dir if required and not supplied.
+    :param authdir: Directory containing authorization files
+
+    :rtype: Sqsh object
+    """
+    def __init__(
+        self,
+        server=None,
+        user=None,
+        passwd=None,
+        database=None,
+        sqshrc=None,
+        authdir="/proj/sot/ska/data/aspect_authorization",
+        **kwargs,
+    ):
+
+        self.server = server or DEFAULT_CONFIG['sybase'].get("server")
+        self.user = user or DEFAULT_CONFIG['sybase'].get("user")
+        self.database = database or DEFAULT_CONFIG['sybase'].get("database")
+        self.passwd = passwd
+        self.sqshrc = sqshrc
+
+        if not Path(SYBASE).exists():
+            raise RuntimeError(f"SYBASE does not exist: {SYBASE}")
+        if not Path(SQSH_BIN).exists():
+            raise RuntimeError(f"sqsh.bin does not exist: {SQSH_BIN}")
+
+        if self.passwd is None and self.sqshrc is None:
+            sqshrc_file = (
+                Path(authdir) / f"sqsh-{self.server}-{self.database}-{self.user}"
+            )
+            if sqshrc_file.exists():
+                self.sqshrc = sqshrc_file
+            else:
+                raise NoPasswordError(
+                    f"None supplied and unable to read inferred sqshrc file {sqshrc_file}"
+                )
+
+    def __enter__(self):
+        """Context manager enter runtime context.  No action required, just return self."""
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """
+        Context manager exit run time context.
+        """
+        pass
+
+
+    def fetch(self, query):
+        """
+        Execute a query and return all returned sql rows a list of lines.
+
+        Note that this doesn't do anything fancy with the types of the return columns.
+        Python Sybase, for example, handled Sybase datetime columns as Python datetimes types
+        and here they are just strings.
+
+        Parameters
+        ----------
+        query : str
+            SQL query to execute (select statements are expected for our use cases)
+
+        Returns
+        -------
+        list of str
+        """
+        envs = ENV_VARS
+        if self.sqshrc is not None:
+            envs = envs + [f"SQSHRC={self.sqshrc}"]
+        cmd = [
+            SQSH_BIN,
+            "-X",
+            "-S",
+            self.server,
+            "-U",
+            self.user,
+            "-D",
+            self.database,
+            "-m",
+            "csv",
+            "-C",
+            query,
+        ]
+        if self.passwd is not None:
+            cmd += ["-P", self.passwd]
```

Comment:
Is this exposing the user password?

Reply:
This is exposing the password on the cmdline I think if for some reason the user decides to use this in this mode, which I wasn't planning on doing but helped in testing.  If we 100% have no other use cases and don't need this to be generic it can go.